### PR TITLE
Use the user's ssh key automatically in ssh examples

### PR DESCRIPTION
**Pull request by justincormack**
This requires moby tool update to support `~` in paths, but
makes everything much nicer.

Signed-off-by: Justin Cormack <justin.cormack@docker.com>

### CODE REVIEW

File: examples/sshd.yml [hunk 1]
```diff
@@ -22,7 +22,8 @@ services:
     image: "linuxkit/sshd:abc1f5e096982ebc3fb61c506aed3ac9c2ae4d55"
 files:
   - path: root/.ssh/authorized_keys
-    contents: '#your ssh key here'
+    source: ~/.ssh/id_rsa.pub
```
Comment (riyazdf) [inline examples/sshd.yml]: I think CI is failing to build this example because it can't find this on its host?

Comment (justincormack) [inline examples/sshd.yml]: hmm, not sure whats best to do about that...

Comment (justincormack) [inline examples/sshd.yml]: maybe will fake it for the tests

Comment (riyazdf) [inline examples/sshd.yml]: faking for the test SGTM

**Comment by ijc** [comment]
Perhaps support something like `only-if-present: true` or maybe `conditional-source: ~/.ssh/foo`?

**Comment by justincormack** [comment]
Added `optional` flag in https://github.com/moby/tool/pull/82 and amended

**Comment by justincormack** [comment]
Ok, this is passing tests now

**Pull request merged**